### PR TITLE
update submitSpecimen derivations and routine

### PR DESCRIPTION
**Pull request by JoeArmani**
Related: https://github.com/episphere/connect/issues/1134

We found bugs in the submitSpecimen routine, which was recently moved from Biospecimen to ConnectFaas.

This PR updates derivations in that routine: blood/urine/mouthwash collected variables and ensures consistency in the accessed data (rm found access to stale specimen data).

### CODE REVIEW

**Review by Gbarra9 — APPROVED** [review]

**Review by bransteitterbr — APPROVED** [review]

File: utils/firestore.js [hunk 1]
```diff
@@ -1304,101 +1304,70 @@ const storeSpecimen = async (data) => {
...
-        }
-        const {processMouthwashEligibility} = require('./validation');
-
-        await db.runTransaction(async transaction => {
-            // @TODO: Honestly we could go through a lot of the helper methods with code overlap
-            // and set them up to allow transactions as a parameter
-            // in order to be called within a transaction without redundant code
+    const { processMouthwashEligibility } = require('./validation');
+    const { buildStreckPlaceholderData, updateBaselineData } = require('./shared');
+    const { checkDerivedVariables } = require('./validation');
```
Comment (we-ai) [inline utils/firestore.js:1309]: Looks like line 1307 and 1309 can be combined.

File: utils/firestore.js [hunk 2]
```diff
@@ -1304,101 +1304,70 @@ const storeSpecimen = async (data) => {
...
+            db.collection('biospecimen').where('token', '==', participantToken)
+                .where(fieldMapping.healthCareProvider.toString(), '==', siteCode)
+                .where(fieldMapping.collectionId.toString(), '==', biospecimenData[fieldMapping.collectionId])
+                .get(),
+        ]);
 
-            // We need to get the token and use it to get the specimen collections and user surveys
-            // Get the token via the equivalent of const token = await getTokenForParticipant(uid); but within a transaction
+        // Validate participant
+        if (participantSnapshot.size !== 1 || specimenCollectionSnapshot.size !== 1) {
```
Comment (we-ai) [inline utils/firestore.js:1333]: Nice checks!

**Review by we-ai — APPROVED** [review]
Looks good. Commented.

**Comment by JoeArmani** [comment]
tested in dev & stage, ready for prod.

**Pull request merged**